### ISSUE TITLE
Script not always invoked on receiving transactions

### ISSUE THREAD

**Issue by seriv** [ethereum/meteor-dapp-wallet#43]
I have tried Developer Preview of Ethereum Wallet: created simple contract-based wallet. Tried sending to it 3 Ethers from my account on hitbtc.com and it worked fine. Later I joined solo mining pool on ethpool.org, and after a while got block reward send to this wallet. It did not show up in balance of the wallet. Later I have sent 100 wei to test if the script will be invoked from my etherbase address to the wallet, and the script was not invoked again. So two transactions are stuck, they did not invoke the script. What I did wrong? Or is it a bug in the wallet's contract script?
Just checked the node is working fine, by sending 1 eth from this contract-based wallet. All 4 transactions can be checked on https://etherscan.io/address/0x8056338e73fde306bb5d9aedec7f4a7b8637c9e7

**Comment by frozeman** [comment]
What you mean with "script invoked"?
Could you provide the transaction hashes of the  transaction which never arrived?

**Comment by seriv** [comment]
Sure!
I ran for a while
```
ethminer -F http://ethpool.org/miner/0x8056338e73fde306bb5d9aedec7f4a7b8637c9e7.rig1/27 -G
```
Ethpool.org has pool address 0x4bb96091ee9d802ed039c4d1a5f6216f90f81b01 After a few days I got reward for the block and nonces, approximately 5 ethers. Precisely, at 10/5/2015 12:00:25 PM transaction http://etherscan.io/tx/0x39b66c720fe5952890091ba3e05f4c85a6121ea8a21c459b66d1fb2e4521c795 for 5.0028683 Ethers with "To:Contract 0x8056338e73fde306bb5d9aedec7f4a7b8637c9e7 Invocation". And I see this transaction on the http://etherscan.io/tx/0x39b66c720fe5952890091ba3e05f4c85a6121ea8a21c459b66d1fb2e4521c795, or https://etherchain.org/tx/0x39b66c720fe5952890091ba3e05f4c85a6121ea8a21c459b66d1fb2e4521c795, with incoming 5.0028 ethers, but balance was not increased.

**Comment by frozeman** [comment]
@obscuren could it be that mining into contracts is not possible?

**Comment by seriv** [comment]
It was not "mining to contracts". Block 337327 was mined by the pool ethpool.org (address 0x4bb96091ee9d802ed039c4d1a5f6216f90f81b01), and reward was sent to address 0x8056338e73fde306bb5d9aedec7f4a7b8637c9e7, which is contract-based wallet. And this transaction did not increase the balance of the recipient address.

I decided to test "mining to contracts", and ran ethminer with coinbase set to this address directly. I got block 388104, and it looks like balance is increased (at least on etherscan.io) for contract, but without any invocation.

**Comment by frozeman** [comment]
ok, so what happened is that the pool set the gas fixed to 21k gas, which is not enough for sending money to the contract, which is ~30k or so..

The funds should still be on the exchanges wallet. People should always provide a little more gas than the minimum. You should bring this to the pools attention

**Comment by frozeman** [comment]
When you mined into the contract, did you saw a transaction appearing? @chriseth does it call the unnamed function also on mined ether?

**Comment by seriv** [comment]
I finally get to the computer where I have meteor-dapp-wallet. And I see the balance is still 2.01 Ethers, and the block reward is not available for spending. Sorry, when I wrote that mining to contracts is working fine, I looked only at etherscan.io and etherchain.org, which both said the balance should be 7.01. It looks like it does not work.

**Comment by frozeman** [comment]
The wallet does check using `web3.eth.getBalance(..)`:
```
> eth.getBalance('0x8056338e73fde306bb5d9aedec7f4a7b8637c9e7')
7010000000000000000
```

So it should show correctly, if you're synced to the latest block.

**Comment by etherchain-org** [comment]
@frozeman Shouldn't the transaction fail if insufficient gas was supplied? I am not sure that the balance is still in the pool's account as the transaction was mined in a block.

**Comment by frozeman** [comment]
Even if the transaction is mined, if the gas is to low, the money would never move to the account, as the state gets reverted..

**Comment by etherchain-org** [comment]
Then at the end who pays for such a transaction if it is stored in the blockchain?

**Comment by seriv** [comment]
Interesting, I checked geth console and saw 
  > eth.getBalance('0x8056338e73fde306bb5d9aedec7f4a7b8637c9e7')
  7010000000000000000
  > eth.blockNumber
  391969

and in the wallet application there was still 2.01 as wallet's balance. 
Then I closed wallet application and restarted geth adding parameter  `--rpccorsdomain "http://localhost:3000"` and unlocked etherbase account. And, after starting wallet's application, I see balance changed to 7.01. Sorry for lot of noise, I should really learn Solidity.
